github/operations.py

```diff
@@ -235,44 +230,41 @@ def fetch_upstream(config, params, *args, **kwargs):
 
 def clone_repository(config, params, *args, **kwargs):
     try:
-        token = config.get('password')
-        g = Github(token)
-        if params.get('repo_type') == 'Organization':
-            repo = g.get_organization(params.get('org')).get_repo(params.get('name'))
+        if params.get('repo_type') == "Organization":
+            url = "https://{0}:{1}@codeload.github.com/{2}/{3}/zip/refs/heads/{4}".format(config.get('username'),
+                                                                                          config.get('password'),
+                                                                                          params.get('org'),
+                                                                                          params.get('name'),
+                                                                                          params.get('branch'))
         else:
-            repo = g.get_user().get_repo(params.get('name'))
-        contents = repo.get_contents("")
-        while contents:
-            file_content = contents.pop(0)
-            if file_content.type == "dir":
-                contents.extend(repo.get_contents(file_content.path))
-            else:
-                completeName = os.path.join('/tmp/{0}/{1}'.format(params.get('name'), file_content.path))
-                wkspFldr = os.path.dirname(completeName)
-                if not os.path.exists(wkspFldr):
-                    os.makedirs(wkspFldr)
-                data = file_content.content
-                if '.png' not in file_content.path:
-                    data = base64.b64decode(data)
-                    with open(completeName, "wb") as file1:
-                        file1.write(data)
-                else:
-                    with Image.open(BytesIO(base64.b64decode(data))) as im:
-                        im.save('/tmp/{0}/{1}'.format(params.get('name'), file_content.path), 'PNG')
+            url = "https://{0}:{1}@codeload.github.com/{2}/{3}/zip/refs/heads/{4}".format(config.get('username'),
+                                                                                          config.get('password'),
+                                                                                          params.get('owner'),
+                                                                                          params.get('name'),
+                                                                                          params.get('branch'))
+        headers = {
+            'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7'}
+        zip_file = '/tmp/github-{0}.zip'.format(params.get('name'))
+        if os.path.isfile(zip_file):
```

Comment:
is this required? opening in "wb" will anyways overwrite

Reply:
done